# Data-Science
## Python
[Efficient Python Tricks and Tools for Data Scientists](https://khuyentran1401.github.io/Efficient_Python_tricks_and_tools_for_data_scientists/README.html)

[Getting started with Python environments (using Conda)](https://towardsdatascience.com/getting-started-with-python-environments-using-conda-32e9f2779307)


## R
[introduction R](https://mgimond.github.io/ES218/Week01.html)

[package development](https://fontikar.github.io/RladiesSYD_DIYRpkg/)    [Github](https://github.com/fontikar/RladiesSYD_DIYRpkg)    [Youtube](https://www.youtube.com/watch?v=4-9jX5F_l8g)

[Building R packages with devtools and usethis | RStudio](https://www.youtube.com/watch?v=EpTkT6Rkgbs)

[Package Development Workshop R-Ladies NYK](https://github.com/jtr13/pkgdev) [DAY1](https://github.com/jtr13/pkgdev/blob/main/Day1_lab.md)   [Day2](https://github.com/jtr13/pkgdev/blob/main/Day2_lab.md)

[Parallel programming](https://berkeley-scf.github.io/tutorial-parallelization/parallel-R)

## Github actions
[Using Github Actions as a job scheduler for R scripts](https://sabeeh.medium.com/using-github-actions-as-a-job-scheduler-for-r-scripts-7b92539372f4)

[Launch an R script using github actions](https://www.rforseo.com/ressources/launch-an-r-script-using-github-actions)

[Deploying ML models in production with R + Github Action](https://www.youtube.com/watch?v=nhTgIyN7NC8&t=2s)

## APIs
[Hosting an R Plumber API using Azure App Service](https://veerlevanleemput.medium.com/hosting-a-r-plumber-api-using-azure-app-service-4e78936787cf)

[A Layman’s Guide for Data Scientists to create APIs in minutes](https://towardsdatascience.com/a-layman-guide-for-data-scientists-to-create-apis-in-minutes-31e6f451cd2f)

[The Power of Parameterized Reports With Plumber](https://www.datalorax.com/post/power-parameterized-reports-plumber/)

## DL
[TF DL](https://dev.mrdbourke.com/tensorflow-deep-learning/)

## ML
[A Gentle Introduction to Matrix Factorization for Machine Learning](https://machinelearningmastery.com/introduction-to-matrix-decompositions-for-machine-learning/)

[Google Crash course](https://developers.google.com/machine-learning/crash-course/ml-intro)

[Making friends with ML](https://www.youtube.com/playlist?list=PLRKtJ4IpxJpDxl0NTvNYQWKCYzHNuy2xG)

[Machine Learning: Step-By-Step](https://towardsdatascience.com/machine-learning-step-by-step-6fbde95c455a)

[Implementing Feature Selection Methods for Machine learning](https://ranasinghiitkgp.medium.com/implementing-feature-selection-methods-for-machine-learning-bfa2e4b4e02)

[When Bayes, Ockham, and Shannon come together to define machine learning](https://towardsdatascience.com/when-bayes-ockham-and-shannon-come-together-to-define-machine-learning-96422729a1ad)

[How to Build an Online Machine Learning App with Python](https://towardsdatascience.com/how-to-build-an-online-machine-learning-app-with-python-c976b2e9f44a)

[Modelling with Tidymodels and Parsnip](https://towardsdatascience.com/modelling-with-tidymodels-and-parsnip-bae2c01c131c)

[Tidymodels](https://rpubs.com/marcelocostamagna/730911)

[pca with tidymodels in R](https://cmdlinetips.com/2020/06/pca-with-tidymodels-in-r/)

[pca example using prcomp in R](https://cmdlinetips.com/2019/04/introduction-to-pca-with-r-using-prcomp/)

[AMLDAfrica tidymodels](https://www.youtube.com/watch?v=KjCfM9oWfng)        [Event link](https://appliedmldays.org/events/amld-africa-2021/workshops/tidy-modelling-in-r)

[Caret and Tidymodels](https://ryjohnson09.netlify.app/post/caret-and-tidymodels/)

[Classification with Tidymodels, Workflows and Recipes](https://www.kirenz.com/post/2021-02-17-r-classification-tidymodels/)

[Microsoft : ML for beginners](https://github.com/microsoft/ML-For-Beginners)

[Jon Krohn ML](https://www.jonkrohn.com/)

[Explain Any Machine Learning Model in Python](https://towardsdatascience.com/shap-explain-any-machine-learning-model-in-python-24207127cad7)

[Dimensionality reduction for machine learning](https://towardsdatascience.com/dimensionality-reduction-for-machine-learning-ef20d8a108d)

[Machine learning complete](https://nbviewer.jupyter.org/github/Nyandwi/machine_learning_complete/tree/main/)

[Feature scaling](https://towardsdatascience.com/all-about-feature-scaling-bcc0ad75cb35)

[Decision Tree — Implemented from scratch](https://towardsdatascience.com/decision-tree-implemented-from-scratch-f11f7ca6aa0)

## Pipelines

[How to Schedule Python Scripts With Cron](https://towardsdatascience.com/how-to-schedule-python-scripts-with-cron-the-only-guide-youll-ever-need-deea2df63b4e)

## MLOPS
[A Practical Guide to MLOps in AWS Sagemaker](https://towardsdatascience.com/a-practical-guide-to-mlops-in-aws-sagemaker-part-i-1d28003f565)

## Kubernetes
[Deploy Dash with Helm on Kubernetes with AWS EKS](https://www.dabbleofdevops.com/blog/deploy-dash-with-helm-on-kubernetes-with-aws-eks)

## Jupyter notebook
[How to Connect to JupyterLab Remotely](https://towardsdatascience.com/how-to-connect-to-jupyterlab-remotely-9180b57c45bb)

[parametrization jupyter notebook](https://jupyter-tutorial.readthedocs.io/en/latest/refactoring/parameterise/index.html)

[From Jupyter Notebook to Deployment — A Straightforward Example](https://towardsdatascience.com/from-jupyter-notebook-to-deployment-a-straightforward-example-1838c203a437)

[From Jupyter Notebook To Scripts](https://towardsdatascience.com/from-jupyter-notebook-to-sc-582978d3c0c)

## Kaggle
[My Journey to Kaggle Master at the Age of 14](https://towardsdatascience.com/my-journey-to-kaggle-master-at-the-age-of-14-e2c42b19c6f7)

## Pandas
[Process ~10M Row Datasets in Milliseconds In This Comprehensive Pandas Speed Guide](https://towardsdatascience.com/process-10m-row-datasets-in-milliseconds-in-this-comprehensive-pandas-speed-guide-5a3125cbb78)

[The illustrated loc and iloc affair](https://towardsdatascience.com/the-illustrated-loc-and-iloc-affair-75b3f378aaf0)

[Python Pandas Tutorial: A Complete Introduction for Beginners](https://www.learndatasci.com/tutorials/python-pandas-tutorial-complete-introduction-for-beginners/)

## Julia

[Building Your First Classification Pipeline In Julia](https://towardsdatascience.com/building-your-first-classification-pipeline-in-julia-47e661c097d7)

## Docker 
[Dockerize Your Dash App](https://towardsdatascience.com/dockerize-your-dash-app-1e155dd1cea3)

# Bioinformatics

## RNAseq
[Dash Apps that Analyze RNA Sequencing Data for Drug & Vaccine Development](https://medium.com/plotly/dash-apps-that-analyze-rna-sequencing-data-for-drug-vaccine-development-1a0ea9763d5)

[Analysis of gene expression timecourse data using maSigPro](https://nsaunders.wordpress.com/2015/05/29/analysis-of-gene-expression-timecourse-data-using-masigpro/)

[Differential Expression - RNA-seq](https://alexslemonade.github.io/refinebio-examples/03-rnaseq/00-intro-to-rnaseq.html)

https://m.blog.naver.com/jinp7/221848536949

https://combine-australia.github.io/RNAseq-R/06-rnaseq-day1.html

https://www.bioconductor.org/packages/devel/workflows/vignettes/rnaseqGene/inst/doc/rnaseqGene.html

https://bioinformatics.uconn.edu/resources-and-events/tutorials-2/rna-seq-tutorial-with-reference-genome/#

https://rpubs.com/laurenblake/384967

https://git.embl.de/provazni/rna-seq-tutorial/-/blob/master/RNAseq_data_analysis.rmd

https://ucdavis-bioinformatics-training.github.io/2018-June-RNA-Seq-Workshop/thursday/DE.html

https://gist.github.com/jdblischak/fdb1745612927252a7633751e5e60bcb

http://www.compbio.dundee.ac.uk/user/pschofield/Projects/teaching_pg/workshops/biocDGE.html

http://www.compbio.dundee.ac.uk/user/pschofield/Projects/teaching_pg/workshops/biocNGS.html

https://www.biostars.org/p/479903/

https://www.nbic.nl/uploads/media/Practical_DiffExpr.pdf

https://github.com/NBISweden/workshop-RNAseq/blob/master/lab_preprocessing.Rmd

https://bioinfotraining.bio.cam.ac.uk/postgraduate/specialized/bioinfo-RNAseq3

https://alexslemonade.github.io/refinebio-examples/03-rnaseq/dimension-reduction_rnaseq_01_pca.html#43_Define_a_minimum_counts_cutoff

https://combine-australia.github.io/RNAseq-R/06-rnaseq-day1.html#Visualising_gene_set_tests:_Barcode_and_enrichment_plots

https://web.azenta.com/a-quick-start-guide-to-rna-seq-data-analysis#step6

https://bioinformatics-core-shared-training.github.io/cruk-summer-school-2018/RNASeq2018/html/02_Preprocessing_Data.nb.html

https://github.com/elsayed-lab/manuscript-shared-rnaseq

https://github.com/MahejabeenNidhi/RNASeqanalysis/blob/master/mousedata/limma.R

https://github.com/lpantano/mypubs/blob/master/srnaseq/mirqc/ready_report.rmd

https://scienceparkstudygroup.github.io/rna-seq-lesson/06-differential-analysis/index.html

https://www.nxn.se/valent/2016/10/3/the-first-steps-in-rna-seq-expression-analysis-single-cell-and-other

https://github.com/snakemake-workflows/single-cell-rna-seq

https://research.stowers.org/cws/CompGenomics/Tutorial/GO_analysis.html

https://www.bioconductor.org/packages/release/bioc/vignettes/conclus/inst/doc/conclus_vignette.pdf

https://nbisweden.github.io/workshop-scRNAseq/labs/compiled/seurat/seurat_05_dge.html

https://icbi-lab.github.io/borst2021/02_filter_data.html

https://ngdc.cncb.ac.cn/cancerscem/browse

https://www.reneshbedre.com/tags/#bioinformatics

https://programmerclick.com/article/98952061357/

https://gtk-teaching.github.io/Microarrays-R/index.html

https://github.com/jdblischak/dc-bioc-limma/blob/master/code/populus.R

https://github.com/MarioniLab/MNN2017/tree/master/PancreasDE

## Pipelines in Bioinformatics
[ISCB Nextflow part one](https://www.youtube.com/watch?v=FNv0b3PeHSw)

[ISCB Nextflow part two](https://www.youtube.com/watch?v=i9pzXcO6w-4)

[GCC2021 Science Training Track](https://galaxyproject.org/events/gcc2021/training/)

[nextflow tutorial](https://yetulaxman.github.io/Biocontainer/tutorials/nextflow_tutorial.html)

[nextflow carpenteries](https://carpentries-incubator.github.io/workflows-nextflow/)

## Pathway analysis
[ISCB Reproducible omics data](https://www.youtube.com/watch?v=uMDIa8Mo6aw)   [Github](https://github.com/BIGCAT-COVID19/ISMB2021-workshop)

## Human Pangenome
[Human Pangenome](https://github.com/human-pangenomics/HPP_Year1_Data_Freeze_v1.0)

[Diversity panel](https://github.com/human-pangenomics/hpgp-data)

## Single cell
https://erilu.github.io/single-cell-rnaseq-analysis/

https://nbisweden.github.io/workshop-scRNAseq/exercises.html

https://hbctraining.github.io/scRNA-seq/lessons/03_SC_quality_control-setup.html

https://smorabit.github.io/tutorials/7_single-cell-essentials/

https://ucdavis-bioinformatics-training.github.io/2019-single-cell-RNA-sequencing-Workshop-UCD_UCSF/data_reduction/Expression_Matrix.html

https://broadinstitute.github.io/2020_scWorkshop/index.html#course-overview

https://swaruplab.bio.uci.edu/tutorial/snRNA-essentials/snRNA-essentials.html

https://satijalab.org/seurat/archive/v3.1/interaction_vignette.html

https://learn.gencore.bio.nyu.edu/single-cell-rnaseq/

[sc multiomics map](https://bookdown.org/ytliu13207/SingleCellMultiOmicsDataAnalysis/seurat-pre-process.html)

https://github.com/satijalab/seurat/issues/432

https://ucdavis-bioinformatics-training.github.io/2017_2018-single-cell-RNA-sequencing-Workshop-UCD_UCB_UCSF/day2/scRNA_Workshop-PART1.html

https://ucdavis-bioinformatics-training.github.io/2020-Advanced_Single_Cell_RNA_Seq/data_analysis/Mapping_Comparison_fixed

https://swaruplab.bio.uci.edu/tutorial/integration/integration_tutorial.html#

https://www.singlecellcourse.org/basic-quality-control-qc-and-exploration-of-scrna-seq-datasets.html

https://satijalab.org/signac/articles/snareseq.html

https://satijalab.org/signac/articles/pbmc_multiomic.html

https://www.bioinformatics.babraham.ac.uk/training/10XRNASeq/seurat_workflow.html

https://broadinstitute.github.io/KrumlovSingleCellWorkshop2020/data-wrangling-scrnaseq-1.html#filtering-low-quality-cells

https://www.stephaniehicks.com/2018-bioinfosummer-scrnaseq/cleaning-the-expression-matrix.html

https://data.bits.vib.be/pub/trainingen/NGSIntro/seurat3_TUTORIAL.html

https://support.parsebiosciences.com/hc/en-us/articles/360053078092-Seurat-Tutorial-65k-PBMCs

https://www.biostars.org/p/9492223/#9492285

http://andrewjohnhill.com/images/posts/2019-5-6-dimensionality-reduction-for-scatac-data/analysis.html

https://satijalab.org/signac/articles/merging.html

https://dburkhardt.github.io/tutorial/preprocessing/

https://github.com/Dragonmasterx87/Merging-multiple-Seurat-Objects/blob/d58ca93118ae1b46c7b227f6ad7ba8b46fb04c4a/Merging%20%3E2%20Seurats.R

https://rpubs.com/Ronlee/SeuratPractice

https://github.com/satijalab/seurat/issues/4955

https://satijalab.org/signac/articles/future.html

https://github.com/vertesy/Seurat.multicore/blob/master/Seurat.Multicore.Examples.R

https://github.com/satijalab/seurat/issues/3126

https://dm5migu4zj3pb.cloudfront.net/manuscripts/136000/136578/JCI136578.sd.pdf

https://satijalab.org/seurat/articles/integration_large_datasets.html

https://github.com/satijalab/seurat/issues/4623

https://rpubs.com/aranyak111/693840

https://rpubs.com/aranyak111/693926

https://satijalab.org/seurat/articles/integration_rpca.html

https://github.com/satijalab/seurat/issues/4419

https://htmlpreview.github.io/?https://github.com/satijalab/seurat.wrappers/blob/master/docs/harmony.html

https://bioconductor.org/books/release/OSCA/book-contents.html

https://satijalab.org/seurat/archive/v3.1/conversion_vignette.html

https://satijalab.org/seurat/archive/v3.1/integration.html#reciprocal-pca

https://bioinfotraining.bio.cam.ac.uk/postgraduate/specialized/bioinfo-RNAseq3

https://github.com/HediaTnani/CellAnnotationTutorial/blob/main/CodingBlocks.Rmd

https://rnabio.org/module-08-scrna/0008/02/01/scRNA/

https://ucdavis-bioinformatics-training.github.io/2020-Intro_Single_Cell_RNA_Seq/data_analysis/scrna_analysis_prepare

https://github.com/satijalab/seurat/issues/3396

https://bioconductor.org/books/release/SingleRBook/

http://bioinformatics.age.mpg.de/presentations-tutorials/presentations/modules/single-cell//bioconductor_tutorial.html

https://icbi-lab.github.io/borst2021/02_filter_data.html

http://bioinformatics.age.mpg.de/presentations-tutorials/presentations/modules/single-cell//bioconductor_tutorial.html

http://oshlacklab.com/combes-organoid-paper/07_Combined_Clustering.html

https://support.10xgenomics.com/single-cell-gene-expression/software/visualization/latest/tutorial-reclustering

https://github.com/chanzuckerberg/scRNA-python-workshop

https://support.10xgenomics.com/single-cell-gene-expression/software/visualization/latest/tutorial-celltypes

https://github.com/haniffalab/Single-cell-RNAseq-data-analysis-bundle

## Deep learning
https://github.com/fchollet/deep-learning-with-python-notebooks

## Carpenteries
https://scienceparkstudygroup.github.io/rna-seq-lesson/01-introduction/index.html

https://scienceparkstudygroup.github.io/rna-seq-lesson/06-differential-analysis/index.html

https://carpentries-incubator.github.io/logistic-regression-public-health/index.html

https://carpentries-incubator.github.io/statistical-thinking-public-health/

https://carpentries-incubator.github.io/machine-learning-novice-sklearn/

https://carpentries-incubator.github.io/deep-learning-intro/index.html

https://carpentries-incubator.github.io/introduction-to-conda-for-data-scientists/

https://carpentries-incubator.github.io/lesson-parallel-python/

https://swcarpentry.github.io/git-novice/

http://swcarpentry.github.io/r-novice-inflammation/15-supp-loops-in-depth/index.html

http://swcarpentry.github.io/r-novice-inflammation/05-cmdline/index.html

https://gtk-teaching.github.io/NGS-intro/

https://ome.github.io/training-docker/

https://carpentries-incubator.github.io/docker-introduction/

https://carpentries-incubator.github.io/docker-introduction/

https://carpentries-incubator.github.io/scrna-seq-analysis/04-Normalization/index.html

https://carpentries-incubator.github.io/bioc-rnaseq/

https://datacarpentry.org/cloud-genomics/

https://datacarpentry.org/shell-genomics/

https://datacarpentry.org/wrangling-genomics/

https://carpentries-incubator.github.io/scrna-seq-analysis/04-Normalization/index.html

https://carpentries-incubator.github.io/docker-introduction/

https://ome.github.io/training-docker/

https://gtk-teaching.github.io/ChIP-seq/

https://gtk-teaching.github.io/NGS-intro/

https://gtk-teaching.github.io/Microarrays-R/index.html

https://carpentries-incubator.github.io/bioc-rnaseq/

https://edcarp.github.io/introduction-to-conda-for-data-scientists/

https://carpentries-incubator.github.io/high-dimensional-stats-r/

https://carpentries-incubator.github.io/snakemake-novice-bioinformatics/

https://carpentries-incubator.github.io/workflows-nextflow/

http://www.hpc-carpentry.org/hpc-python/

https://blog.byronjsmith.com/snakemake-analysis.html

https://snakemake.readthedocs.io/en/stable/tutorial/basics.html

## Batch effects
https://evayiwenwang.github.io/Managing_batch_effects/adjust.html#sva

http://jtleek.com/genstats/inst/doc/02_13_batch-effects.html

## Heatmaps
https://jokergoo.github.io/ComplexHeatmap-reference/book/introduction.html

## Mouse to human gene names
https://www.r-bloggers.com/2016/10/converting-mouse-to-human-gene-names-with-biomart-package/

## purrr
https://jennybc.github.io/purrr-tutorial/ls02_map-extraction-advanced.html

https://byuistats.github.io/M335/parallel_furrr.html

https://bookdown.org/rdpeng/rprogdatascience/parallel-computation.html

## Targets
https://books.ropensci.org/targets/hpc.html

## PCA
https://www.biostars.org/p/479903/

https://rpubs.com/crazyhottommy/PCA_MDS

https://bioconductor.org/packages/release/bioc/vignettes/PCAtools/inst/doc/PCAtools.html

https://cran.r-project.org/web/packages/ggfortify/vignettes/plot_pca.html

## Shiny
https://engineering-shiny.org/optimizing-shiny-code.html

## Functional annotation
https://nbisweden.github.io/workshop-RNAseq/2111/lab_functional.html#21_enrichR

https://www.biostars.org/p/339934/

## small RNAseq
https://github.com/nf-core/smrnaseq/blob/master/bin/edgeR_miRBase.r

## Structural variants
https://www.bioconductor.org/packages/release/bioc/vignettes/StructuralVariantAnnotation/inst/doc/vignettes.html

## deseq2
https://github.com/compbiocore/DEseq_workshop_2_2021/blob/main/deseq_workshop_2.ipynb

https://github.com/compbiocore/rnaseq_workshop_2021/blob/main/deseq_1.ipynb

https://github.com/compbiocore/ccv_bootcamp_bioinfo/blob/main/annot_resources.ipynb


## DSL2
https://telatin.github.io/microbiome-bioinformatics/Nextflow-start/

## bioinformatics tutorials
https://ucdavis-bioinformatics-training.github.io/

## Rmarkdown mail
https://pkgs.rstudio.com/blastula/

## 10x genomics

https://www.10xgenomics.com/videos?query=&page=1

https://support.10xgenomics.com/single-cell-gene-expression/software/visualization/latest/tutorial-reclustering

https://support.10xgenomics.com/single-cell-gene-expression/software/visualization/latest/tutorial-celltypes


## Data vizualization
https://miqg.github.io/practical_tools_for_quick_data_visualization/README.html

## Multiomics
https://biofam.github.io/MOFA2/tutorials.html

## Rmarkdown
https://www.earthdatascience.org/courses/earth-analytics/multispectral-remote-sensing-modis/add-variables-to-rmarkdown-report/

https://www.datalorax.com/post/power-parameterized-reports-plumber/

https://towardsdatascience.com/parameters-in-r-markdown-ade0cfac0c9b

https://towardsdatascience.com/api-interaction-with-r-46bf1d4f4fe6

## Snakemake
https://github.com/ctb/2019-snakemake-ucdavis/blob/master/tutorial.md

## Bioinformatics teaching
https://github.com/ASPteaching

https://edcarp.github.io/Ed-DaSH/ed_dash_team.html
